### DISCUSSION TITLE
Continue processing after response.

### DISCUSSION THREAD

**Discussion by FernandoKGA** [vercel/vercel#4314] [General] [answered]
Can the Serverless Functions continue processing after sending an response?
Example: I have to do some complex thing and I don't to make the domain that is requesting keeps waiting. So I send back an 202 (Accepted) or 204 (No Content), and keep doing my thing until finish it.

Is this even possible?

```js
module.exports = async (req, res) => {
  //get some data from req
  res.status(202).send(null);
  //keep doing my thing
  return;
}
```

Update: I'm trying to call some URLs after the execution of the response, the function doesn't call them, does Now closes all connections after sending the response?

**Comment by coetry** [comment, marked as answer]
> does Now closes all connections after sending the response?

Yes. As soon as you send a response, all processing comes to a halt.

**Reply by FernandoKGA** [reply]
This is described in the Serverless Functions Docs?

**Reply by styfle** [reply]
Yes, its documented here https://vercel.com/docs/v2/platform/limits#streaming-responses

**Reply by FernandoKGA** [reply]
Thanks @styfle ! Sorry, probably I didn't pay attention to this part...

**Comment by ScreamZ** [comment]
@coetry @styfle Any way to interact with AWS config from a lambda?

`context.callbackWaitsForEmptyEventLoop = false;`

Could be useful as described here: https://docs.atlas.mongodb.com/best-practices-connecting-to-aws-lambda/

Thanks

**Reply by styfle** [reply]
You can use the AWS Lambda API directly like this: https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage/aws-lambda-api

**Reply by styfle** [reply]
But our current API already sets `callbackWaitsForEmptyEventLoop` to false, see #4548.

**Comment by y-nk** [comment]
@styfle there's a usecase where i'd like to be able to use the vercel functions to power a slack bot ; slack bots have a timeout of 3s to send a response, after that it's required to send a post request somewhere else. i'd like to know if/how i could achieve this with the current flow of vercel functions, do you have any idea?

**Reply by styfle** [reply]
You can use [Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions) to stream the response so that you can send the partial response before 3 seconds and then finish up once you have more data to stream.

**Reply by y-nk** [reply]
I will try following the link you provided. Thanks @styfle !

**Reply by y-nk** [reply]
@styfle it looks quite broken here :)

I've setup a quick repo to try and here are my feedbacks.

1. ~I can't test locally bc the function throws an (undocumented) error message (reported [here](https://github.com/vercel/edge-runtime/issues/209))~ solved.

2. Logs are not accessible in the inspect panel in Vercel's dashboard after running `vercel deploy` and invoking a function

3. Whatever I do it seems that a browser does not understand the stream and computes/buffers the response anyway so the whole thing is kinda pointless.

If you have a time, please have a look and tell me where i'm wrong at using edge functions.

- Repo: https://github.com/y-nk/streaming-at-edge
- Deployment: https://streaming-at-edge-y-nk.vercel.app/api/stream

I can add you to the Vercel's project if necessary.

**Reply by styfle** [reply]
@y-nk That example is only sending one byte each time. Try sending more data. This works:

- PR: https://github.com/y-nk/streaming-at-edge/pull/1
- Deployment: https://streaming-at-edge-git-fork-styfle-patch-1-y-nk.vercel.app/api/stream

**Reply by y-nk** [reply]
@styfle thanks a ton for this

**Reply by blakeross** [reply]
@y-nk Is this example code still available anywhere? Thanks!

**Reply by styfle** [reply]
Looks like the repo above was deleted.

Here's the official streaming example:

https://github.com/vercel/examples/blob/3dca5c2be358fb9b0d8c297713883f02ab2f1682/edge-functions/streams/pages/api/01-simple.ts

**Reply by y-nk** [reply]
you can try @styfle fork if it's still there too
